### PR TITLE
Add initial verification builder

### PR DESCRIPTION
**Pull request by franzTobiasDLR**
Add a simple builder that updates the requirement status depending on the state of the different verification methods. Intendet to be extended by customized verification methods later...

Provides infrastructure for the verification builder, which will be customized later.

### CODE REVIEW

File: de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/provider/RequirementsAttributeLabelProvider.java [hunk 1]
```diff
@@ -177,6 +184,35 @@ public Image getColumnImage(Object object, int columnIndex) {
...
+	/**
+	 * Get the icon for the requirement status
+	 * @param statusValue the current status value
+	 * @return the image
+	 */
+	protected Image getStatusIcon(String statusValue) {
+		if (statusValue == null) {
+			return null;
+		}
+		String imagePath = ICON_STATUS_OPEN_PATH;
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/provider/RequirementsAttributeLabelProvider.java:197]: Would it be possible to make the structure here a bit simple by extracting a getImagePath(statusValue) method?

File: de.dlr.sc.virsat.model.extension.requirements/src/de/dlr/sc/virsat/model/extension/requirements/verification/build/RequirementsStatusUpdater.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		subMonitor.beginTask("Updating Status on Requirments in Specification " + specification.getName(), verificationTaskNumber);
+		for (RequirementObject requirementObject : specification.getRequirements()) {
+			updateRequirementStatus(requirementObject, subMonitor);
+			subMonitor.worked(1);
+		}
+	}
+	
+	/**
+	 * Update the status of the different requirement objects in a specification 
+	 * @param requirementObject	the requirement object to verify
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.model.extension.requirements/src/de/dlr/sc/virsat/model/extension/requirements/verification/build/RequirementsStatusUpdater.java:44]: weird whitespace

File: de.dlr.sc.virsat.model.extension.requirements/src/de/dlr/sc/virsat/model/extension/requirements/verification/build/RequirementsStatusUpdater.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	 * @param requirement the requirement to be updated
+	 */
+	protected void computeNewStatus(Requirement requirement) {
+		boolean inProgress = false;
+		boolean partlyNonCompliant = false;
+		boolean partlyCompliant = false;
+		boolean fullyCompliant = false;
+		
+		for (IVerification verification : requirement.getVerification()) {
+			if (verification.getStatus() != null) {
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.model.extension.requirements/src/de/dlr/sc/virsat/model/extension/requirements/verification/build/RequirementsStatusUpdater.java:74]: Maybe extract the `verfification.getStatus()` into a variable so we dont have to repeat it in all if checks.

File: de.dlr.sc.virsat.model.extension.requirements/src/de/dlr/sc/virsat/model/extension/requirements/verification/build/RequirementsStatusUpdater.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+				} else if (verification.getStatus().equals(IVerification.STATUS_Open_NAME)) {
+					inProgress = true;
+				} else if (verification.getStatus().equals(IVerification.STATUS_PartialCompliant_NAME)) {
+					partlyCompliant = true;
+				}
+			}
+		}
+		
+		if (inProgress) {
+			editingDomain.getCommandStack().execute(requirement.setStatus(editingDomain, Requirement.STATUS_Open_NAME));
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.model.extension.requirements/src/de/dlr/sc/virsat/model/extension/requirements/verification/build/RequirementsStatusUpdater.java:88]: We could extract a command variable and execute that at the end.

File: de.dlr.sc.virsat.model.extension.requirements/src/de/dlr/sc/virsat/model/extension/requirements/verification/build/RequirementsVerificationBuilder.java [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+		
+	}
+	
+	/**
+	 * Run the verification on a SEI within the project
+	 * @param seiBean the SEI Bean
+	 * @param monitor the progress monitor
+	 */
+	protected void buildSei(BeanStructuralElementInstance seiBean, IProgressMonitor monitor) {
+		List<CategoryAssignment> caSpecifications = seiBean.getStructuralElementInstance().getCategoryAssignments().stream().
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.model.extension.requirements/src/de/dlr/sc/virsat/model/extension/requirements/verification/build/RequirementsVerificationBuilder.java:137]: Any reason we cannot do `List<Requirementsspecification> specifications = seiBean.getAll(RquirementsSpecification.class)`

**Review by SaMuellerDLR — CHANGES_REQUESTED** [review]
No major issues.

File: de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/provider/RequirementsAttributeLabelProvider.java (hunk 1 above)
Comment (franzTobiasDLR) [reply de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/provider/RequirementsAttributeLabelProvider.java:207]: Not sure why it doesn't show outdated... I applyed your suggestion however... Although I don't really see a simplification here :D

**Review by SaMuellerDLR — APPROVED** [review]
Approved!

**Review by JAmmermann-DLR — APPROVED** [review]

**Pull request merged**